corgi/tasks/prod_defs.py

```
@@ -159,12 +197,12 @@ def parse_product_version(
     pd_product_version: dict[str, Any], product: Product, product_node: ProductNode
 ):
     """Parse the product versions from ps_modules in product-definitions.json"""
-    pd_product_streams = pd_product_version.pop("product_streams", [])
-    name = pd_product_version.pop("id")
+    pd_product_streams = pd_product_version.get("product_streams", [])
```

Doing .get() here silently hides an error due to missing keys. See other comment for details, but let's fix this one too, unless we need this for some reason. If this really is needed / some products don't define the product_streams key, I'd rather fix the prod-defs data to be valid, and let Corgi fail with an error when it's not.

---

We use product_streams value again at the end of the look to check for deleted streams.

---

Ack, see my reply in the other comment.